src/components/date_picker/super_date_picker/super_date_picker.tsx

```diff
@@ -56,6 +64,13 @@ export type EuiSuperDatePickerProps = CommonProps & {
   commonlyUsedRanges?: DurationRange[];
   customQuickSelectPanels?: QuickSelectPanel[];
 
+  /**
+   * A function that allows the Quick Select panels to render in a custom order
```

Comment:
Docs feedback - I think we should highlight this specifically as a `render prop` since that's a keyword in React's documentation and is more easily googleable for folks who may be familiar w/ the concept.

I also added in more verbose notes about the flexibility a render prop allows

```suggestion
   * An optional render prop function that allows customizing the display of the Quick Select menu.
   * This function passes all default quick select panels within an object, allowing you to
   * re-order panels, omit certain panels entirely, or pass in your own fully custom content.
```